### ISSUE TITLE
Discovery and Query issues

### ISSUE THREAD

**Issue by MazdakFarzone** [mjansson/mdns#24]
Building the latest version makes devices respond correctly. But the software doesn't handle the query responses. The responses are observable in Wireshark, but not by the software

The only responses I'm getting is the response from the client which sends the discovery query. In other words it is only responding to itself with the services it has available. 

Tried simple printf-debugging but couldn't find any clues at all. It seems like the responses aren't even acquired in the `mdns_query_recv`

**Comment by mjansson** [comment]
The problem comes from my confusion reading the spec. What this library implements is basically simple one-shot queries and a service to answer such queries.

One-shot queries should, however, be sent from a high-number ephemeral port and answered over unicast. So I reverted the change to always use port 5353 to comply with the RFC regarding one-shot queries and unicast responses.

The service part is only made to service such one-shot queries and might not be fully standard compliant (yet).

To implement a full mDNS service with continuous queries, answers and announcements is a larger effort, I've added a tracking for this in #25
